### PR TITLE
4th milestone comments and likes

### PR DESCRIPTION
**Pull request by IvanDerlich**
### **This milestone**

- Comment and Likes model with validations and associations
- Unit tests and integrations tests for comment and likes model
- Add factories for comment and likes
- Add tests for feature 'user post comments'
- Add tests for feature 'user likes post' 
- Requirements to create a comment: be signed in, from 0 to 300 characters accepted
- Requirements to create a like: be signed in, you can only like once per post, you cannot dislike
- Seed comments table
- Seed likes table
- Likes and Comment controller
- Add comment form partial view
- Edit post form partial view
- Add Like link in post
- Edit post partial to show comments

### CODE REVIEW

File: app/models/comment.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+class Comment < ApplicationRecord
+  belongs_to :user
+  belongs_to :post
+  validates :text, :user_id, presence: true
```
Comment (mosaaleb) [inline app/models/comment.rb]: - [ ] We don't need to validate `user_id`, `belongs_to :user` adds presence validation to `user_id` by default.

File: app/models/like.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+class Like < ApplicationRecord
+  belongs_to :user
+  belongs_to :post
+  # validates that there is no other post with the same combination of user_id and post_id
+  validates_uniqueness_of :user, scope: :post
```
Comment (mosaaleb) [inline app/models/like.rb]: - Nice job handling likes duplication :clap:

File: app/controllers/comments_controller.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+class CommentsController < ApplicationController
+  before_action :authenticate_user!, only: %i[create]
+  before_action :find_post
+
+  def create
+    @comment = @post.comments.build(comment_params)
+    @comment.user_id = current_user.id
```
Comment (mosaaleb) [inline app/controllers/comments_controller.rb]: - [ ] **[OPTIONAL]** We can also use `current_user.comments << @comment`

File: app/controllers/comments_controller.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+class CommentsController < ApplicationController
+  before_action :authenticate_user!, only: %i[create]
+  before_action :find_post
+
+  def create
```
Comment (mosaaleb) [inline app/controllers/comments_controller.rb]: - [ ] Let's also handle the case if comments didn't save due to errors.

File: app/controllers/comments_controller.rb (hunk 3 above)
Comment (IvanDerlich) [reply app/controllers/comments_controller.rb]: https://github.com/IvanDerlich/TheNewFacebook/pull/6/commits/6f30d551cc002832c5e79c2e935f2d61ee360d0f

File: app/models/like.rb (hunk 2 above)
Comment (IvanDerlich) [reply app/models/like.rb]: https://github.com/IvanDerlich/TheNewFacebook/pull/6/commits/d59b588bdc7d82b5a16c4876da983d2a37e97b05

File: app/controllers/comments_controller.rb (hunk 4 above)
Comment (IvanDerlich) [reply app/controllers/comments_controller.rb]: https://github.com/IvanDerlich/TheNewFacebook/pull/6/commits/a4ebe61c61d822ff85be0179d46401397df0b0d5

**Pull request merged**